Training24/Test.c

```diff
@@ -1,14 +1,161 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
+// Kalaimagal V P
 // ------------------------------------------------------------------------------------------------
 // Test.c
-// Program on main branch.
+// Program on A5 branch.
+// Test.c - The program sorts integers with insertion sort and searches for numbers using binary search, 
+// with options for test cases and user input.
 // ------------------------------------------------------------------------------------------------
 #include <stdio.h>
-#include "Header.h"
+#include <ctype.h>    // For tolower
+#include <stdlib.h>   // For strtol and realloc
+#include <conio.h>    // For getch()
+#include <stdbool.h>
+#include "header.h"
 
+// ANSI escape codes for colors
+#define RESET   "\033[0m"
+#define SKYBLUE "\033[36m"
+#define GREEN   "\033[32m"
+#define YELLOW  "\033[33m"
+#define RED     "\033[31m"
+
+/// <summary>Function to clear the console screen.</summary>
+void ClearScreen () {
+#ifdef _WIN32
+   system ("cls");           // Windows-specific command to clear the screen
+#endif
+}
+
+/// <summary>Function to display the elements of an array.</summary>
+void DisplayArray (IndexedElement arr[], int size) {
+   printf ("| ");
+   for (int i = 0; i < size; i++) printf ("%d  ", arr[i].value);
+   printf ("|\n");
+}
+
+/// <summary>Function to check if the array is sorted.</summary>
+bool IsSorted (IndexedElement arr[], int size) {
+   for (int i = 1; i < size; i++) if (arr[i - 1].value > arr[i].value) return false; // Not sorted
+   return true; // Sorted
+}
+/// <summary>Checks if the input string is a valid integer (digits only).</summary>
+int IsValidInteger (const char* str) {
+   if (str == NULL || *str == '\n') return 0;
+   for (int i = 0; str[i] != '\0' && str[i] != '\n'; i++) if (!isdigit (str[i])) return 0;

```

Comment:
Method should accept arithmetic symbols like '-' and '+', isdigit method will allow only numeric characters
![err](https://github.com/user-attachments/assets/9b0cd35a-e498-40d7-8f9c-6af8a3298cd2)